src/Messages/Interactions/TSMessage.m

```diff
@@ -102,6 +102,7 @@ - (instancetype)initWithTimestamp:(uint64_t)timestamp
     _expiresInSeconds = expiresInSeconds;
     _expireStartedAt = expireStartedAt;
     [self updateExpiresAt];
+    _receivedAtDate = [NSDate date];
```

Comment:
> Does this do the right thing for sync transcript messages (e.g. messages that you sent via desktop) received late? I guess if the server takes care to deliver us the messages in the order it received them?

Yes, there's a single queue so this should work.  

> By tracking local clock time of creation, instead of envelope timestamp...

Right, we were already doing this for incoming messages and this just generalizes that to all `TSMessages`.

> One nit: can we call it createdAtDate since the existing name is so close to the similar but different receivedAt?

I'm happy to change the name, but I feel like `createdAtDate` is a bit misleading.  "Created" evokes (at least for me) when the message was created by the sender.